rust/src/crypto.rs

```diff
@@ -327,4 +329,57 @@ macro_rules! impl_signature {
     };
 }
 
-impl_signature!(Ed25519Signature, Vec<u8>, crypto::Ed25519);
\ No newline at end of file
+impl_signature!(Ed25519Signature, Vec<u8>, crypto::Ed25519);
+
+macro_rules! impl_hash_type {
```

Comment:
From the cddl we'd just have all the hash types all as `Vec<u8>` which isn't the most helpful since they're always fixed-sized, so this macro is just to quickly add on whatever hash types we need and have them do size-checks/etc.